Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.Ballistic.cs

```diff
@@ -62,6 +63,19 @@ private void OnBallisticInteractUsing(EntityUid uid, BallisticAmmoProviderCompon
 
     private void OnBallisticAfterInteract(EntityUid uid, BallisticAmmoProviderComponent component, AfterInteractEvent args)
     {
+
+        // Wrong magazine
+        if (HasComp<GunComponent>(args.Target) && HasComp<ItemSlotsComponent>(args.Target))
+        {
+            Popup(
+                    Loc.GetString("gun-ballistic-transfer-invalid",
+                        ("ammoEntity", uid),
+                        ("targetEntity", args.Target.Value)),
+                    uid,
+                    args.User);
+            return;
+        }
+
```

Comment:
Shouldn't this use itemslots popups instead?

Reply:
I'm not sure what you mean, ItemSlotsSystem doesn't seem to have a popup class, it uses SharedPopupSystem. Is that what you refer to?

Reply:
ItemSlots have a bunch of popup fails for interactions failed.

Reply:
Imagine my utter mental collapse when I realised that this can indeed be achieved by a SINGLE LINE on the yaml, if I just add whitelistFailPopup to the magazine itemslot. I assume that's what you meant. However, this will cause the same fixed message warning for ANY item being clicked on the gun, while my initial solution would only trigger if a magazine  -but the wrong one- is used. Is this a distinction that matters? I guess it *would* be more consistent and less prone to any potential breakage if ItemSlotSystem's built-in method was used. Should I go ahead and strip out my code and just apply WhitelistFailPopups to every gun?

Reply:
I removed the code from the system and specified ItemSlotComponent's WhitelistFailPopup datafield for all existing guns and baseguns that use magazines